pkg/deploy/registry/deployconfig/strategy.go

```diff
@@ -0,0 +1,81 @@
+package deployconfig
+
+import (
+	"fmt"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/fielderrors"
+
+	"github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/deploy/api/validation"
+)
+
+// strategy implements behavior for DeploymentConfig objects
+type strategy struct {
+	runtime.ObjectTyper
+	kapi.NameGenerator
+}
+
+// Strategy is the default logic that applies when creating and updating DeploymentConfig objects.
+var Strategy = strategy{kapi.Scheme, kapi.SimpleNameGenerator}
+
+func (strategy) NamespaceScoped() bool {
+	return true
+}
+
+// AllowCreateOnUpdate is false for DeploymentConfig objects.
+func (strategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+// PrepareForCreate clears fields that are not allowed to be set by end users on creation.
+func (strategy) PrepareForCreate(obj runtime.Object) {
+	_ = obj.(*api.DeploymentConfig)
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (strategy) PrepareForUpdate(obj, old runtime.Object) {
+	_ = obj.(*api.DeploymentConfig)
+}
+
+// Validate validates a new policy.
+func (strategy) Validate(ctx kapi.Context, obj runtime.Object) fielderrors.ValidationErrorList {
+	return validation.ValidateDeploymentConfig(obj.(*api.DeploymentConfig))
+}
+
+// ValidateUpdate is the default update validation for an end user.
+func (strategy) ValidateUpdate(ctx kapi.Context, obj, old runtime.Object) fielderrors.ValidationErrorList {
+	// TODO: distinguish updates from create, for now, this is preserving old implementation behavior
+	return validation.ValidateDeploymentConfig(obj.(*api.DeploymentConfig))
```

Comment:
We must explicitly validate ObjectMeta update.  Otherwise someone can change the UID.